lib/src/stack/coap_chain.dart

```diff
@@ -265,17 +265,17 @@ class CoapChain<TChain, TFilter, TNextFilter>
       }
       e = e.nextEntry as CoapEntry<TFilter?, TNextFilter>?;
     }
-    return e as CoapIEntry<TFilter, TNextFilter>;
+    return e! as CoapIEntry<TFilter, TNextFilter>;
```

Comment:
I think the `!` could be removed here

```suggestion
    return e as CoapIEntry<TFilter, TNextFilter>;
```

Reply:
No: https://dart-lang.github.io/linter/lints/cast_nullable_to_non_nullable.html

Reply:
Oh, I see, thanks for the clarification!